testdata/p4_16_errors_outputs/table-entries-optional-2-bmv2.p4-stderr

```diff
@@ -0,0 +1,12 @@
+table-entries-optional-2-bmv2.p4(71): [--Wwarn=mismatch] warning: 0x100: value does not fit in 8 bits
+            (0x100, default): a_with_control_params(2);
+             ^^^^^
+table-entries-optional-2-bmv2.p4(73): [--Wwarn=mismatch] warning: 0x10000: value does not fit in 16 bits
+            (default, 0x10000): a_with_control_params(3);
+                      ^^^^^^^
+table-entries-optional-2-bmv2.p4(69): [--Werror=legacy] error: &&& invalid key expression
```

Comment:
are the output files up-to-date? I would have expected this error message: https://github.com/p4lang/p4c/pull/2146/files#diff-7d9ca24e25fbe2b12ec7df5a3e6f2bc3R125

Reply:
That is weird.  The Travis tests all pass, which suggests the output files are up to date, if they are being checked by the test scripts properly.  But the line numbers look off, so not sure why the tests are passing.  I will try regenerating the output files on my machine and see if anything changes.

Reply:
Double-checked, and I was wrong about the line numbers -- the line numbers in these warning and error messages do match what they should be from the P4_16 source file.

Also regenerated the output files from the code in this PR, and it matches what you see here.  I haven't dug hard to see why this error message is given instead of the one you linked to.

I guess the question is: is it worth digging more?  I was just happy to get an error message at all for that line :-)

Reply:
I just checked and this is because the error is given by the P4Runtime serializer and not the bmv2 backend (since the test is run with `--p4runtime-files`). We could make the error more helpful, but that can be a future improvement. After all, the same error would be output if you were to use a `Mask` expression for an exact match.

Reply:
So, no changes required :)